src/sidebar/components/test/util.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Wait for a condition to evaluate to a truthy value.
+ *
+ * @param {() => any} condition - Function that returns a truthy value when some condition is met
+ * @param {number} timeout - Max delay in milliseconds to wait
+ * @param {string} what - Description of condition that is being waited for
+ * @return {Promise<any>} - Result of the `condition` function
+ */
+export async function waitFor(
```

Comment:
This feels like one of those things that can be shared between lms and client -- like the rpc stuff. Gotta flush that out at a later time..